cmd/pbstatuser/main.go

```diff
@@ -20,25 +30,162 @@ import (
-	// would be a really good here: https://gist.github.com/fracasula/b579d52daf15426e58aa133d0340ccb0
+	// Fetch authentication from Sources
+	authentication, err := sourcesClient.GetAuthentication(ctx, sourceId)
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			logger.Warn().Msgf("not found error: %s", err)
+			return
+		}
+		logger.Warn().Msgf("could not get authentication: %s", err)
+		return
+	}
+
+	s := SourceInfo{
+		Authentication: *authentication,
+		SourceID:       sourceId,
+		Identity:       ctxval.Identity(ctx),
+	}
+
+	switch authentication.ProviderType {
+	case models.ProviderTypeAWS:
+		chAws <- s
+	case models.ProviderTypeAzure:
+		chAzure <- s
+	case models.ProviderTypeGCP:
+		chGcp <- s
+	case models.ProviderTypeNoop:
+	case models.ProviderTypeUnknown:
+		logger.Warn().Msg("authentication provider type is unknown")
+	}
+}
+
+func checkSourceAvailabilityAzure(ctx context.Context) {
+	defer processingWG.Done()
+
+	for s := range chAzure {
+		sr := kafka.SourceResult{
+			SourceID: s.SourceID,
+			Identity: s.Identity,
+		}
+		// TODO: check if source is avavliable - WIP
+		sr.Status = kafka.StatusAvaliable
+		chSend <- sr
+	}
+}
+
+func checkSourceAvailabilityAWS(ctx context.Context) {
+	logger := ctxval.Logger(ctx)
+	defer processingWG.Done()
+
+	for s := range chAws {
+		sr := kafka.SourceResult{
+			SourceID: s.SourceID,
+			Identity: s.Identity,
+		}
+		_, err := clients.GetEC2Client(ctx, &s.Authentication, "")
+		if err != nil {
+			sr.Status = kafka.StatusUnavailable
+			sr.Err = err
+			logger.Warn().Msgf("could not get aws assumed client %s", err)
+			chSend <- sr
+		} else {
+			sr.Status = kafka.StatusAvaliable
+			chSend <- sr
+		}
+	}
+}
+
+func checkSourceAvailabilityGCP(ctx context.Context) {
+	logger := ctxval.Logger(ctx)
+	defer processingWG.Done()
+
+	for s := range chGcp {
+		sr := kafka.SourceResult{
+			SourceID: s.SourceID,
+			Identity: s.Identity,
+		}
+		gcpClient, err := clients.GetGCPClient(ctx, &s.Authentication)
+		if err != nil {
+			sr.Status = kafka.StatusUnavailable
+			sr.Err = err
+			logger.Warn().Msgf("could not get gcp client %s", err)
+			chSend <- sr
+			continue
+		}
+		_, err = gcpClient.ListAllRegions(ctx)
+		if err != nil {
+			sr.Status = kafka.StatusUnavailable
+			sr.Err = err
+			logger.Warn().Msgf("could not list gcp regions %s", err)
+			chSend <- sr
+		} else {
+			sr.Status = kafka.StatusAvaliable
+			chSend <- sr
+		}
+	}
+}
+
+func sendResults(ctx context.Context) {
+	logger := ctxval.Logger(ctx)
+	messages := make([]*kafka.GenericMessage, 0, 100)
+	defer senderWG.Done()
+
+	for sr := range chSend {
+		ctx = ctxval.WithIdentity(ctx, sr.Identity)
+		msg, err := sr.GenericMessage(ctx)
+		if err != nil {
+			logger.Warn().Msgf("could not generate generic message %s", err)
+			continue
+		}
+		messages = append(messages, &msg)
+	}
+	err := kafka.Send(ctx, messages...)
+	if err != nil {
+		logger.Warn().Msgf("could not send message through kafka %s", err)
+	}
 }
```

Comment:
This function is not right. This will panic when you receive 101th message from the channel. There are two options how to implement this:

1. Create and send messages on the fly as they are received from the channel, this is less performant solution, however, good for the first version.
2. Do whatever we do in the sending endpoint - a buffer with sending messages either when a buffer is full or every 2 seconds whatever comes first.